### ISSUE TITLE
Inconsistency in `SampledValue.value` shape between `DeterministicVariable` and `DistributionalRV`

### ISSUE THREAD

**Issue by damonbayer** [CDCgov/PyRenew#386]
`DeterministicVariable`s return `SampledValue(value=self.value, t_start=self.t_start, t_unit=self.t_unit,)`, while `DistributionalRV`s return `SampledValue(jnp.atleast_1d(sample), t_start=self.t_start, t_unit=self.t_unit,)`. We should be consistent about using or not using `jnp.atleast_1d`, but I'm not sure which.

I think I am lightly in favor of always calling `jnp.atleast_1d`.

Thoughts @dylanhmorris @gvegayon?

**Comment by dylanhmorris** [comment]
Also lightly in favor of `atleast_1d`

**Comment by damonbayer** [comment]
I misread. Both of these functions use `jnp.atleast1d`. `DeterministicVariables` just does it in the constructor, rather than the `sample` call.

**Comment by dylanhmorris** [comment]
I changed my mind while working on #328. I now think we should avoid calling `atleast_1d` in both of these cases. I think this will play better with `scan` when constructing a 1d array from scalars. 

Details: `scan()` stacks scalars to create an array of dimension `(n_iterations,)`, but stacks 1d arrays to create a 2d array of dimension `(n_iterations, 1)`. 2d arrays with a unit last axis don't behave like 1d arrays.

**Comment by gvegayon** [comment]
I think adding `atleast_1d` provides consistency. In some cases, some functions may break when instead of an array of size one they receive a scalar, e.g., scalars don't have shape, which is expected across `pyrenew`.

**Comment by dylanhmorris** [comment]
Clarification: talking about `jax` scalars here, not standard python `float`s. `jax` scalars do have shape; it's `()`.

```python
import jax.numpy as jnp
a = jnp.array(0.25)
a.shape
```

**Comment by damonbayer** [comment]
Closed in https://github.com/CDCgov/PyRenew/pull/380
